From 7841bfec112c342476c1602b3bae4dbf3d69fa4d Mon Sep 17 00:00:00 2001
From: Ke Qinghua <qinghua.ke@freescale.com>
Date: Fri, 7 Mar 2014 10:22:35 +0800
Subject: [PATCH 290/310] ENGR00302221-1:CTS test faild for security setting
 and test Vroot

Add graphics_device dir/file access rules

Signed-off-by: Ke Qinghua <qinghua.ke@freescale.com>
---
 evk_6sl/BoardConfig.mk      | 4 ++++
 evk_6sl/sepolicy/app.te     | 4 ++++
 sabresd_6dq/BoardConfig.mk  | 5 +++++
 sabresd_6dq/sepolicy/app.te | 4 ++++
 4 files changed, 17 insertions(+)
 create mode 100644 evk_6sl/sepolicy/app.te
 create mode 100644 sabresd_6dq/sepolicy/app.te

diff --git a/evk_6sl/BoardConfig.mk b/evk_6sl/BoardConfig.mk
index 2716c22..1e06d1b 100644
--- a/evk_6sl/BoardConfig.mk
+++ b/evk_6sl/BoardConfig.mk
@@ -79,3 +79,7 @@ NUM_FRAMEBUFFER_SURFACE_BUFFERS := 3
 
 BOARD_KERNEL_CMDLINE := console=ttymxc0,115200 init=/init androidboot.console=ttymxc0 video=mxc_elcdif_fb:bpp=32 androidboot.hardware=freescale csi
 TARGET_BOOTLOADER_CONFIG := mx6sl:mx6sl_evk_android_config
+BOARD_SEPOLICY_DIRS := \
+       device/fsl/evk_6sl/sepolicy
+BOARD_SEPOLICY_UNION := \
+       app.te
diff --git a/evk_6sl/sepolicy/app.te b/evk_6sl/sepolicy/app.te
new file mode 100644
index 0000000..92d8fa6
--- /dev/null
+++ b/evk_6sl/sepolicy/app.te
@@ -0,0 +1,4 @@
+# Grant GPU access to all processes started by Zygote.
+# They need that to render the standard UI.
+allow domain    graphics_device:chr_file rw_file_perms;
+allow domain    graphics_device:dir { write create add_name setattr mounton search };
diff --git a/sabresd_6dq/BoardConfig.mk b/sabresd_6dq/BoardConfig.mk
index 7c7a67a..8f4b48e 100755
--- a/sabresd_6dq/BoardConfig.mk
+++ b/sabresd_6dq/BoardConfig.mk
@@ -109,3 +109,8 @@ IMX6_CONSUMER_IR_HAL := true
 TARGET_BOOTLOADER_CONFIG := 6q:mx6q_sabresd_android_config 6dl:mx6dl_sabresd_android_config
 
 
+BOARD_SEPOLICY_DIRS := \
+       device/fsl/sabresd_6dq/sepolicy
+
+BOARD_SEPOLICY_UNION := \
+       app.te
diff --git a/sabresd_6dq/sepolicy/app.te b/sabresd_6dq/sepolicy/app.te
new file mode 100644
index 0000000..92d8fa6
--- /dev/null
+++ b/sabresd_6dq/sepolicy/app.te
@@ -0,0 +1,4 @@
+# Grant GPU access to all processes started by Zygote.
+# They need that to render the standard UI.
+allow domain    graphics_device:chr_file rw_file_perms;
+allow domain    graphics_device:dir { write create add_name setattr mounton search };
-- 
1.8.0

